tests/catboost/test_catboost_model_export.py

```diff
@@ -0,0 +1,288 @@
+
+
+@pytest.fixture
+def reg_model():
+    model = cb.CatBoostRegressor(**MODEL_PARAMS)
+    X, y = get_iris()
+    return ModelWithData(model=model.fit(X, y), inference_dataframe=X)
+
+
+@pytest.fixture
+def model_path(tmpdir):
+    return tmpdir.join("model").strpath
+
+
+@pytest.fixture
+def custom_env(tmpdir):
+    conda_env_path = os.path.join(tmpdir.strpath, "conda_env.yml")
+    _mlflow_conda_env(conda_env_path, additional_pip_deps=["catboost", "pytest"])
+    return conda_env_path
+
+
+@pytest.mark.large
+@pytest.mark.parametrize("model_type", ["CatBoost", "CatBoostClassifier", "CatBoostRegressor"])
+def test_init_model(model_type):
+    model = mlflow.catboost._init_model(model_type)
+    assert model.__class__.__name__ == model_type
+
+
+@pytest.mark.large
+def test_init_model_throws_for_invalid_model_type():
+    with pytest.raises(TypeError, match="Invalid model type"):
+        mlflow.catboost._init_model("unsupported")
+
+
+@pytest.mark.large
+def test_model_save_load(cb_model, model_path):
+    model, inference_dataframe = cb_model
+    mlflow.catboost.save_model(cb_model=model, path=model_path)
+
+    loaded_model = mlflow.catboost.load_model(model_uri=model_path)
+    np.testing.assert_array_almost_equal(
+        model.predict(inference_dataframe), loaded_model.predict(inference_dataframe),
+    )
+
+    loaded_pyfunc = pyfunc.load_pyfunc(model_uri=model_path)
+    np.testing.assert_array_almost_equal(
+        loaded_model.predict(inference_dataframe), loaded_pyfunc.predict(inference_dataframe),
+    )
+
+
+@pytest.mark.large
+def test_log_model_logs_model_type(cb_model):
+    with mlflow.start_run():
+        artifact_path = "model"
+        mlflow.catboost.log_model(cb_model.model, artifact_path)
+        model_uri = mlflow.get_artifact_uri(artifact_path)
+
+    flavor_conf = Model.load(model_uri).flavors["catboost"]
+    assert "model_type" in flavor_conf
+    assert flavor_conf["model_type"] == cb_model.model.__class__.__name__
+
+
+# Supported serialization formats:
+# https://catboost.ai/docs/concepts/python-reference_catboost_save_model.html
+SUPPORTS_DESERIALIZATION = ["cbm", "coreml", "json", "onnx"]
+save_formats = SUPPORTS_DESERIALIZATION + ["python", "cpp", "pmml"]
+
+
+@pytest.mark.large
+@pytest.mark.parametrize("save_format", save_formats)
+def test_log_model_logs_save_format(reg_model, save_format):
+    with mlflow.start_run():
+        artifact_path = "model"
+        mlflow.catboost.log_model(reg_model.model, artifact_path, format=save_format)
+        model_uri = mlflow.get_artifact_uri(artifact_path)
+
+    flavor_conf = Model.load(model_uri).flavors["catboost"]
+    assert "save_format" in flavor_conf
+    assert flavor_conf["save_format"] == save_format
+
+    if save_format in SUPPORTS_DESERIALIZATION:
+        mlflow.catboost.load_model(model_uri)
+    else:
+        with pytest.raises(cb.CatBoostError, match="deserialization not supported or missing"):
```

Comment:
this case how to load model ?

Reply:
What do you mean?

Reply:
the case save format is "python", "cpp", "pmml", then it can save model, but load model raise error ?

Reply:
Yes, deserialization is not supported for several formats

https://catboost.ai/docs/concepts/python-reference_catboost_save_model.html